website/docs/concepts/html/elements.mdx

```diff
@@ -140,6 +140,47 @@ html! {
 
 If the attribute is set to `None`, the attribute will not be set in the DOM.
 
+## Children
+
+Most HTML elements accept arbitrary HTML as children, however, there is a set of them that doesn't accept any children at all.
+These elements are called _void_ elements, and they are:
+
+-   `<area />`
+-   `<base />`
+-   `<base />`
+-   `<br />`
+-   `<col />`
+-   `<embed />`
+-   `<hr />`
+-   `<img />`
+-   `<input />`
+-   `<link />`
+-   `<meta />`
+-   `<param />`
+-   `<source />`
+-   `<track />`
+-   `<wbr />`
+-   `<textarea />`
+
+Attempting to provide children to these elements will result in a compilation error or, if the element tag is chosen dynamically, in a panic.
+
+### The case of `<textarea>`
+
+The `<textarea>` element is special; The modern HTML specification states that children of `<textarea>` define its default value, however in Yew it's specified differently.
+Instead of writing
+
+```html
+<textarea>{"default value"}</textarea>
+```
+
+Which would fail to compile, it's customary to write
+
+```html
+<textarea ~defaultValue="default value"></textarea>
```

Comment:
```suggestion
<textarea defaultvalue="default value" />
```

I am not in favour of promoting the property syntax for elements. It should only be reserved for cases where it is niche and should not / cannot be handled by the renderer (e.g.: custom elements).

I think we should make a special handing to assign `defaultvalue` (`html!` attribute) to `value` at the first time it is mounted.

I feel that there might be edge cases for `~defaultValue` (property) when it is combined with Suspense as the element might be re-mounted.

Reply:
And what would be the disadvantages to mimicking the actual HTML specification and make `<textarea>` accept children?

Reply:
The childNodes for an `HTMLTextAreaElement` indicates its default value and not value.

Once the user starts to type in an textarea, normal means Yew uses to manipulate the DOM (e.g.: `HTMLTextAreaElement.insertBefore`) has no effect anymore. Manipulating childNodes does not reflect on the page.

Reply:
Then we could, for example, not do anything with `<textarea>`'s children after it's inserted into the DOM, i.e. not reconcile it, which also might improve performance, besides, we already have special handling for `<textarea>`, it even has a separate variant for it in the `VTagInner` enum. It's just that going against the HTML spec would only unnecessarily complicate the framework from the perspective of the users, I believe that when it comes to HTML elements, we should strive to keep their usage in Yew as close to their actual definition in the standard as possible.

Reply:
> It's just that going against the HTML spec would only unnecessarily complicate the framework from the perspective of the users

The [specification](https://html.spec.whatwg.org/multipage/form-elements.html#dom-textarea-defaultvalue) says that HtmlTextAreaElement's default value can be assigned via the `defaultValue` property. Since elements in Yew are created via `document.createElement`, we are not against the specification.

Specifically not reconciling textarea is only going to make users confused as it is not obvious to novice users that textarea is only defining its default value and value updates will need to be done via value attribute. It was also not obvious to me until after some previous discussion about making textarea to use children prop happened in the past.

This pull request makes it a compile error, which is a better user experience IMO.

In addition, React and Vue both do not accept children for textarea. So we are consistent with the behaviour of major frameworks.
 
React: https://react.dev/reference/react-dom/components/textarea#props
Vue: https://vuejs.org/guide/essentials/forms.html#multiline-text

Reply:
I'm curious about the use case where the textarea is needed and isn't controlled